content/docs/introduction/faq.md

```diff
@@ -153,3 +163,40 @@ should take less than a minute under normal circumstances. See [crash recovery](
 
 You have run into a bug of ZFS on Linux. See [issue #484](https://github.com/prometheus/prometheus/issues/484)
 for details. Upgrading to ZFS on Linux v0.6.4 should fix the issue.
+
+## Implementation
+
+### Why are all sample values 64-bit floats? I want integers.
+
+We restrained ourselves to 64-bit floats to simplify the design. The
+[IEEE 754 double-precision binary floating-point format](http://en.wikipedia.org/wiki/Double-precision_floating-point_format)
+supports integer precision for values up to 2<sup>53</sup>. Supporting
+native 64 bit integers would (only) help if you need integer precision
+above 2<sup>53</sup> but below 2<sup>63</sup>. In principle, support
+for different sample value types (including some kind of big integer,
+supporting even more than 64 bit) could be implemented, but it is not
+really a priority right now.
+
+### Why does Prometheus use a custom storage backend rather than [some other storage system]? Isn't the "one file per time series" approach killing performance?
```

Comment:
Funny enough, the deletion speed wasn't really an issue in practice. (Originally, we had one _directory_ per time series, with files in it, but ever since we got rid of that, we had no problems when blowing away a massive amount of files.) Perhaps XFS (which we are using) is faster in deleting files?